src/bin/ion/commands/beta/symtab/filter.rs

```diff
@@ -52,50 +51,58 @@ impl IonCliCommand for SymtabFilterCommand {
             for input_file in input_file_names {
                 let file = File::open(input_file.as_str())
                     .with_context(|| format!("Could not open file '{}'", &input_file))?;
-
-                let mmap = unsafe {
-                    MmapOptions::new()
-                        .map(&file)
-                        .with_context(|| format!("Could not mmap '{}'", input_file))?
-                };
-
-                // Treat the mmap as a byte array.
-                let ion_data: &[u8] = &mmap[..];
-                let raw_reader = RawBinaryReader::new(ion_data);
-                let mut system_reader = SystemReader::new(raw_reader);
-                omit_user_data(ion_data, &mut system_reader, &mut output, lift_requested)?;
+                let mut system_reader = SystemReader::new(AnyEncoding, file);
+                filter_out_user_data(&mut system_reader, &mut output, lift_requested)?;
             }
         } else {
-            bail!("this command does not yet support reading from STDIN")
+            let mut system_reader = SystemReader::new(AnyEncoding, io::stdin().lock());
+            filter_out_user_data(&mut system_reader, &mut output, lift_requested)?;
         }
 
         output.flush()?;
         Ok(())
     }
 }
 
-pub fn omit_user_data(
-    ion_data: &[u8],
-    reader: &mut SystemReader<RawBinaryReader<&[u8]>>,
+pub fn filter_out_user_data(
+    reader: &mut SystemReader<AnyEncoding, impl IonInput>,
     output: &mut Box<dyn Write>,
     lift_requested: bool,
-) -> IonResult<()> {
+) -> Result<()> {
     loop {
-        match reader.next()? {
-            SystemStreamItem::VersionMarker(major, minor) => {
-                output.write_all(&[0xE0, major, minor, 0xEA])?;
+        match reader.next_item()? {
+            SystemStreamItem::VersionMarker(marker) => {
+                output.write_all(marker.span().bytes())?;
             }
-            SystemStreamItem::SymbolTableValue(IonType::Struct) => {
-                if !lift_requested {
-                    output.write_all(reader.raw_annotations_bytes().unwrap_or(&[]))?;
+            SystemStreamItem::SymbolTable(symtab) => {
+                let Some(raw_value) = symtab.as_value().raw() else {
+                    // This symbol table came from a macro expansion; there are no encoded bytes
+                    // to pass through.
+                    bail!("found an ephemeral symbol table, which is not yet supported")
```

Comment:
Is "ephemeral" going to be an official-ish term for this case?

Reply:
That's the phrase I've been using for it, but admittedly mostly to myself :)

If it resonates with you, then I'd say let's do it. 👍